README.md
##MaterialPaperColors 0.2.0
---

I wrote this package to have easy access to googles material paper background and text color palette via ranges  
as shown on http://www.google.com/design/spec/style/color.html - 30.06.2014 

---

###Example

###access single color direct
    
      input:
      String color = Red["500"];
      or
      List color = RedT["500"];
      output:
      "#e51c23"
      or
      ["#e51c23", "rgba(255, 255, 255, 0.87)"]

###access color palette direct

    input:
      List<Map<String,String>> colorPalette = colorPalette;
      or
      List<Map<String,List<String>>> colorPaletteT = colorPaletteT;
    output:
      [Red, Pink, Purple, DeepPurple, Indigo, Blue, LightBlue, Cyan, Teal, Green,
       LightGreen, Lime, Yellow, Amber, Orange, DeepOrange, Brown, Grey, BlueGrey]
      or
      [RedT, PinkT, PurpleT, DeepPurpleT, IndigoT, BlueT, LightBlueT, CyanT, TealT, GreenT,
       LightGreenT, LimeT, YellowT, AmberT, OrangeT, DeepOrangeT, BrownT, GreyT, BlueGreyT]
      
###get random color via function

    input:
      String randColor = getRandomColor("500");
      List randColorT = getRandomColorT("500");
    output:
      "#e91e63" // eg. Pink
      or 
      ["#e91e63", "rgba(0, 0, 0, 0.87)"] // eg. PinkT

###get random color palette via function

    input:
      Map randColorPalette = getRandomColorPalette();
      Map randColorPaletteT = getRandomColorPaletteT();
    output:
      Blue // Map  
      or 
      BlueT // Map 
    
###supported colors 
as shown on http://www.google.com/design/spec/style/color.html - 30.06.2014

     Red= {
      "50": "#fde0dc",
      "100": "#f9bdbb",
      "200": "#f69988",
      "300": "#f36c60",
      "400": "#e84e40",
      "500": "#e51c23",
      "600": "#dd191d",
      "700": "#d01716",
      "800": "#c41411",
      "900": "#b0120a",
      "A100": "#ff7997",
      "A200": "#ff5177",
      "A400": "#ff2d6f",
      "A700": "#e00032"
     }
     Pink= {
      "50": "#fce4ec",
      "100": "#f8bbd0",
      "200": "#f48fb1",
      "300": "#f06292",
      "400": "#ec407a",
      "500": "#e91e63",
      "600": "#d81b60",
      "700": "#c2185b",
      "800": "#ad1457",
      "900": "#880e4f",
      "A100": "#ff80ab",
      "A200": "#ff4081",
      "A400": "#f50057",
      "A700": "#c51162"
     }
     Purple= {
      "50": "#f3e5f5",
      "100": "#e1bee7",
      "200": "#ce93d8",
      "300": "#ba68c8",
      "400": "#ab47bc",
      "500": "#9c27b0",
      "600": "#8e24aa",
      "700": "#7b1fa2",
      "800": "#6a1b9a",
      "900": "#4a148c",
      "A100": "#ea80fc",
      "A200": "#e040fb",
      "A400": "#d500f9",
      "A700": "#aa00ff"
     }
     DeepPurple = {
      "50": "#ede7f6",
      "100": "#d1c4e9",
      "200": "#b39ddb",
      "300": "#9575cd",
      "400": "#7e57c2",
      "500": "#673ab7",
      "600": "#5e35b1",
      "700": "#512da8",
      "800": "#4527a0",
      "900": "#311b92",
      "A100": "#b388ff",
      "A200": "#7c4dff",
      "A400": "#651fff",
      "A700": "#6200ea"
     }
     Indigo = {
      "50": "#e8eaf6",
      "100": "#c5cae9",
      "200": "#9fa8da",
      "300": "#7986cb",
      "400": "#5c6bc0",
      "500": "#3f51b5",
      "600": "#3949ab",
      "700": "#303f9f",
      "800": "#283593",
      "900": "#1a237e",
      "A100": "#8c9eff",
      "A200": "#536dfe",
      "A400": "#3d5afe",
      "A700": "#304ffe"
     }
     Blue={
      "50": "#e7e9fd",
      "100": "#d0d9ff",
      "200": "#afbfff",
      "300": "#91a7ff",
      "400": "#738ffe",
      "500": "#5677fc",
      "600": "#4e6cef",
      "700": "#455ede",
      "800": "#3b50ce",
      "900": "#2a36b1",

CHANGELOG.md
VERSION
-------
    
    0.2.0 -added functions to access a randomColorPalette additionally to only a single randomColor
          -fixed textColors
    0.1.0 -changed to top level functions
    0.0.2 -added textColors via T Maps and function
    

lib/material_paper_colors.dart
library material_paper_colors;

import 'dart:math';

final Map<String, String> Red = {
  "50": "#fde0dc",
  "100": "#f9bdbb",
  "200": "#f69988",
  "300": "#f36c60",
  "400": "#e84e40",
  "500": "#e51c23",
  "600": "#dd191d",
  "700": "#d01716",
  "800": "#c41411",
  "900": "#b0120a",
  "A100": "#ff7997",
  "A200": "#ff5177",
  "A400": "#ff2d6f",
  "A700": "#e00032"
},
    Pink = {
      "50": "#fce4ec",
      "100": "#f8bbd0",
      "200": "#f48fb1",
      "300": "#f06292",
      "400": "#ec407a",
      "500": "#e91e63",
      "600": "#d81b60",
      "700": "#c2185b",
      "800": "#ad1457",
      "900": "#880e4f",
      "A100": "#ff80ab",
      "A200": "#ff4081",
      "A400": "#f50057",
      "A700": "#c51162"
    },
    Purple = {
      "50": "#f3e5f5",
      "100": "#e1bee7",
      "200": "#ce93d8",
      "300": "#ba68c8",
      "400": "#ab47bc",
      "500": "#9c27b0",
      "600": "#8e24aa",
      "700": "#7b1fa2",
      "800": "#6a1b9a",
      "900": "#4a148c",
      "A100": "#ea80fc",
      "A200": "#e040fb",
      "A400": "#d500f9",
      "A700": "#aa00ff"
    },
    DeepPurple = {
      "50": "#ede7f6",
      "100": "#d1c4e9",
      "200": "#b39ddb",
      "300": "#9575cd",
      "400": "#7e57c2",
      "500": "#673ab7",
      "600": "#5e35b1",
      "700": "#512da8",
      "800": "#4527a0",
      "900": "#311b92",
      "A100": "#b388ff",
      "A200": "#7c4dff",
      "A400": "#651fff",
      "A700": "#6200ea"
    },
    Indigo = {
      "50": "#e8eaf6",
      "100": "#c5cae9",
      "200": "#9fa8da",
      "300": "#7986cb",
      "400": "#5c6bc0",
      "500": "#3f51b5",
      "600": "#3949ab",
      "700": "#303f9f",
      "800": "#283593",
      "900": "#1a237e",
      "A100": "#8c9eff",
      "A200": "#536dfe",
      "A400": "#3d5afe",
      "A700": "#304ffe"
    },
    Blue = {
      "50": "#e7e9fd",
      "100": "#d0d9ff",
      "200": "#afbfff",
      "300": "#91a7ff",
      "400": "#738ffe",
      "500": "#5677fc",
      "600": "#4e6cef",
      "700": "#455ede",
      "800": "#3b50ce",
      "900": "#2a36b1",
      "A100": "#a6baff",
      "A200": "#6889ff",
      "A400": "#4d73ff",
      "A700": "#4d69ff"
    },
    LightBlue = {
      "50": "#e1f5fe",
      "100": "#b3e5fc",
      "200": "#81d4fa",
      "300": "#4fc3f7",
      "400": "#29b6f6",
      "500": "#03a9f4",
      "600": "#039be5",
      "700": "#0288d1",
      "800": "#0277bd",
      "900": "#01579b",
      "A100": "#80d8ff",
      "A200": "#40c4ff",
      "A400": "#00b0ff",
      "A700": "#0091ea"
    },
    Cyan = {
      "50": "#e0f7fa",
      "100": "#b2ebf2",
      "200": "#80deea",
      "300": "#4dd0e1",
      "400": "#26c6da",
      "500": "#00bcd4",
      "600": "#00acc1",
      "700": "#0097a7",
      "800": "#00838f",
      "900": "#006064",
      "A100": "#84ffff",
      "A200": "#18ffff",
      "A400": "#00e5ff",
      "A700": "#00b8d4"
    },
    Teal = {
      "50": "#e0f2f1",
      "100": "#b2dfdb",
      "200": "#80cbc4",
      "300": "#4db6ac",
      "400": "#26a69a",
      "500": "#009688",
      "600": "#00897b",
      "700": "#00796b",
      "800": "#00695c",
      "900": "#004d40",
      "A100": "#a7ffeb",
      "A200": "#64ffda",
      "A400": "#1de9b6",
      "A700": "#00bfa5"
    },
    Green = {
      "50": "#d0f8ce",
      "100": "#a3e9a4",
      "200": "#72d572",
      "300": "#42bd41",
      "400": "#2baf2b",
      "500": "#259b24",
      "600": "#0a8f08",
      "700": "#0a7e07",
      "800": "#056f00",
      "900": "#0d5302",
      "A100": "#a2f78d",
      "A200": "#5af158",
      "A400": "#14e715",
      "A700": "#12c700"
    },
    LightGreen = {
      "50": "#f1f8e9",
      "100": "#dcedc8",
      "200": "#c5e1a5",
      "300": "#aed581",
      "400": "#9ccc65",
      "500": "#8bc34a",
      "600": "#7cb342",
      "700": "#689f38",
      "800": "#558b2f",
      "900": "#33691e",
      "A100": "#ccff90",
      "A200": "#b2ff59",
      "A400": "#76ff03",
      "A700": "#64dd17"
    },
    Lime = {
      "50": "#f9fbe7",
      "100": "#f0f4c3",
      "200": "#e6ee9c",
      "300": "#dce775",
      "400": "#d4e157",
      "500": "#cddc39",
      "600": "#c0ca33",
      "700": "#afb42b",
      "800": "#9e9d24",
      "900": "#827717",
      "A100": "#f4ff81",
      "A200": "#eeff41",
      "A400": "#c6ff00",
      "A700": "#aeea00"
    },
    Yellow = {
      "50": "#fffde7",
      "100": "#fff9c4",
      "200": "#fff59d",
      "300": "#fff176",
      "400": "#ffee58",
      "500": "#ffeb3b",
      "600": "#fdd835",
      "700": "#fbc02d",
      "800": "#f9a825",
      "900": "#f57f17",
      "A100": "#ffff8d",
      "A200": "#ffff00",
      "A400": "#ffea00",
      "A700": "#ffd600"
    },
    Amber = {
      "50": "#fff8e1",
      "100": "#ffecb3",
      "200": "#ffe082",
      "300": "#ffd54f",
      "400": "#ffca28",
      "500": "#ffc107",
      "600": "#ffb300",
      "700": "#ffa000",
      "800": "#ff8f00",
      "900": "#ff6f00",
      "A100": "#ffe57f",
      "A200": "#ffd740",
      "A400": "#ffc400",
      "A700": "#ffab00"
    },
    Orange = {
      "50": "#fff3e0",
      "100": "#ffe0b2",
      "200": "#ffcc80",
      "300": "#ffb74d",
      "400": "#ffa726",
      "500": "#ff9800",
      "600": "#fb8c00",
      "700": "#f57c00",
      "800": "#ef6c00",
      "900": "#e65100",
      "A100": "#ffd180",
      "A200": "#ffab40",
      "A400": "#ff9100",
      "A700": "#ff6d00"
    },
    DeepOrange = {
      "50": "#fbe9e7",
      "100": "#ffccbc",
      "200": "#ffab91",
      "300": "#ff8a65",
      "400": "#ff7043",
      "500": "#ff5722",
      "600": "#f4511e",
      "700": "#e64a19",
      "800": "#d84315",
      "900": "#bf360c",
      "A100": "#ff9e80",
      "A200": "#ff6e40",
      "A400": "#ff3d00",
      "A700": "#dd2c00"
    },
    Brown = {
      "50": "#efebe9",
      "100": "#d7ccc8",
      "200": "#bcaaa4",
      "300": "#a1887f",
      "400": "#8d6e63",
      "500": "#795548",
      "600": "#6d4c41",
      "700": "#5d4037",
      "800": "#4e342e",
      "900": "#3e2723"
    },
    Grey = {
      "50": "#fafafa",
      "100": "#f5f5f5",
      "200": "#eeeeee",
      "300": "#e0e0e0",
      "400": "#bdbdbd",
      "500": "#9e9e9e",
      "600": "#757575",
      "700": "#616161",
      "800": "#424242",
      "900": "#212121"
    },
    BlueGrey = {
      "50": "#eceff1",
      "100": "#cfd8dc",
      "200": "#b0bec5",
      "300": "#90a4ae",
      "400": "#78909c",
      "500": "#607d8b",
      "600": "#546e7a",
      "700": "#455a64",
      "800": "#37474f",
      "900": "#263238"
    };


final Map<String, List<String>> RedT = {
  "50": ["#fde0dc", "rgba(0, 0, 0, 0.87)"],
  "100": ["#f9bdbb", "rgba(0, 0, 0, 0.87)"],
  "200": ["#f69988", "rgba(0, 0, 0, 0.87)"],
  "300": ["#f36c60", "rgba(0, 0, 0, 0.87)"],
  "400": ["#e84e40", "rgba(0, 0, 0, 0.87)"],
  "500": ["#e51c23", "#ffffff"],
  "600": ["#dd191d", "#ffffff"],
  "700": ["#d01716", "#ffffff"],
  "800": ["#c41411", "rgba(255, 255, 255, 0.87)"],
  "900": ["#b0120a", "rgba(255, 255, 255, 0.87)"],
  "A100": ["#ff7997", "rgba(0, 0, 0, 0.87)"],
  "A200": ["#ff5177", "#ffffff"],
  "A400": ["#ff2d6f", "#ffffff"],
  "A700": ["#e00032", "#ffffff"]
},
    PinkT = {
      "50": ["#fce4ec", "rgba(0, 0, 0, 0.87)"],
      "100": ["#f8bbd0", "rgba(0, 0, 0, 0.87)"],
      "200": ["#f48fb1", "rgba(0, 0, 0, 0.87)"],
      "300": ["#f06292", "rgba(0, 0, 0, 0.87)"],
      "400": ["#ec407a", "rgba(0, 0, 0, 0.87)"],
      "500": ["#e91e63", "#ffffff"],
      "600": ["#d81b60", "#ffffff"],
      "700": ["#c2185b", "rgba(255, 255, 255, 0.87)"],
      "800": ["#ad1457", "rgba(255, 255, 255, 0.87)"],
      "900": ["#880e4f", "rgba(255, 255, 255, 0.87)"],
      "A100": ["#ff80ab", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#ff4081", "#ffffff"],
      "A400": ["#f50057", "#ffffff"],
      "A700": ["#c51162", "#ffffff"]
    },
    PurpleT = {
      "50": ["#f3e5f5", "rgba(0, 0, 0, 0.87)"],
      "100": ["#e1bee7", "rgba(0, 0, 0, 0.87)"],
      "200": ["#ce93d8", "rgba(0, 0, 0, 0.87)"],
      "300": ["#ba68c8", "#ffffff"],
      "400": ["#ab47bc", "#ffffff"],
      "500": ["#9c27b0", "rgba(255, 255, 255, 0.87)"],
      "600": ["#8e24aa", "rgba(255, 255, 255, 0.87)"],
      "700": ["#7b1fa2", "rgba(255, 255, 255, 0.87)"],
      "800": ["#6a1b9a", "rgba(255, 255, 255, 0.87)"],
      "900": ["#4a148c", "rgba(255, 255, 255, 0.87)"],
      "A100": ["#ea80fc", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#e040fb", "#ffffff"],
      "A400": ["#d500f9", "#ffffff"],
      "A700": ["#aa00ff", "#ffffff"]
    },
    DeepPurpleT = {
      "50": ["#ede7f6", "rgba(0, 0, 0, 0.87)"],
      "100": ["#d1c4e9", "rgba(0, 0, 0, 0.87)"],
      "200": ["#b39ddb", "rgba(0, 0, 0, 0.87)"],
      "300": ["#9575cd", "#ffffff"],
      "400": ["#7e57c2", "#ffffff"],
      "500": ["#673ab7", "rgba(255, 255, 255, 0.87)"],
      "600": ["#5e35b1", "rgba(255, 255, 255, 0.87)"],
      "700": ["#512da8", "rgba(255, 255, 255, 0.87)"],
      "800": ["#4527a0", "rgba(255, 255, 255, 0.87)"],
      "900": ["#311b92", "rgba(255, 255, 255, 0.87)"],
      "A100": ["#b388ff", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#7c4dff", "#ffffff"],
      "A400": ["#651fff", "rgba(255, 255, 255, 0.87)"],
      "A700": ["#6200ea", "rgba(255, 255, 255, 0.87)"]
    },
    IndigoT = {
      "50": ["#e8eaf6", "rgba(0, 0, 0, 0.87)"],
      "100": ["#c5cae9", "rgba(0, 0, 0, 0.87)"],
      "200": ["#9fa8da", "rgba(0, 0, 0, 0.87)"],
      "300": ["#7986cb", "#ffffff"],
      "400": ["#5c6bc0", "#ffffff"],
      "500": ["#3f51b5", "rgba(255, 255, 255, 0.87)"],
      "600": ["#3949ab", "rgba(255, 255, 255, 0.87)"],
      "700": ["#303f9f", "rgba(255, 255, 255, 0.87)"],
      "800": ["#283593", "rgba(255, 255, 255, 0.87)"],
      "900": ["#1a237e", "rgba(255, 255, 255, 0.87)"],
      "A100": ["#8c9eff", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#536dfe", "#ffffff"],
      "A400": ["#3d5afe", "#ffffff"],
      "A700": ["#304ffe", "rgba(255, 255, 255, 0.87)"]
    },
    BlueT = {
      "50": ["#e7e9fd", "rgba(0, 0, 0, 0.87)"],
      "100": ["#d0d9ff", "rgba(0, 0, 0, 0.87)"],
      "200": ["#afbfff", "rgba(0, 0, 0, 0.87)"],
      "300": ["#91a7ff", "rgba(0, 0, 0, 0.87)"],
      "400": ["#738ffe", "rgba(0, 0, 0, 0.87)"],
      "500": ["#5677fc", "#ffffff"],
      "600": ["#4e6cef", "#ffffff"],
      "700": ["#455ede", "#ffffff"],
      "800": ["#3b50ce", "rgba(255, 255, 255, 0.87)"],
      "900": ["#2a36b1", "rgba(255, 255, 255, 0.87)"],
      "A100": ["#a6baff", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#6889ff", "#ffffff"],
      "A400": ["#4d73ff", "#ffffff"],
      "A700": ["#4d69ff", "#ffffff"]
    },
    LightBlueT = {
      "50": ["#e1f5fe", "rgba(0, 0, 0, 0.87)"],
      "100": ["#b3e5fc", "rgba(0, 0, 0, 0.87)"],
      "200": ["#81d4fa", "rgba(0, 0, 0, 0.87)"],
      "300": ["#4fc3f7", "rgba(0, 0, 0, 0.87)"],
      "400": ["#29b6f6", "rgba(0, 0, 0, 0.87)"],
      "500": ["#03a9f4", "#ffffff"],
      "600": ["#039be5", "#ffffff"],
      "700": ["#0288d1", "#ffffff"],
      "800": ["#0277bd", "#ffffff"],
      "900": ["#01579b", "rgba(255, 255, 255, 0.87)"],
      "A100": ["#80d8ff", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#40c4ff", "rgba(0, 0, 0, 0.87)"],
      "A400": ["#00b0ff", "rgba(0, 0, 0, 0.87)"],
      "A700": ["#0091ea", "#ffffff"]
    },
    CyanT = {
      "50": ["#e0f7fa", "rgba(0, 0, 0, 0.87)"],
      "100": ["#b2ebf2", "rgba(0, 0, 0, 0.87)"],
      "200": ["#80deea", "rgba(0, 0, 0, 0.87)"],
      "300": ["#4dd0e1", "rgba(0, 0, 0, 0.87)"],
      "400": ["#26c6da", "rgba(0, 0, 0, 0.87)"],
      "500": ["#00bcd4", "#ffffff"],
      "600": ["#00acc1", "#ffffff"],
      "700": ["#0097a7", "#ffffff"],
      "800": ["#00838f", "#ffffff"],
      "900": ["#006064", "rgba(255, 255, 255, 0.87)"],
      "A100": ["#84ffff", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#18ffff", "rgba(0, 0, 0, 0.87)"],
      "A400": ["#00e5ff", "rgba(0, 0, 0, 0.87)"],
      "A700": ["#00b8d4", "rgba(0, 0, 0, 0.87)"]
    },
    TealT = {
      "50": ["#e0f2f1", "rgba(0, 0, 0, 0.87)"],
      "100": ["#b2dfdb", "rgba(0, 0, 0, 0.87)"],
      "200": ["#80cbc4", "rgba(0, 0, 0, 0.87)"],
      "300": ["#4db6ac", "rgba(0, 0, 0, 0.87)"],
      "400": ["#26a69a", "rgba(0, 0, 0, 0.87)"],
      "500": ["#009688", "#ffffff"],
      "600": ["#00897b", "#ffffff"],
      "700": ["#00796b", "#ffffff"],
      "800": ["#00695c", "rgba(255, 255, 255, 0.87)"],
      "900": ["#004d40", "rgba(255, 255, 255, 0.87)"],
      "A100": ["#a7ffeb", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#64ffda", "rgba(0, 0, 0, 0.87)"],
      "A400": ["#1de9b6", "rgba(0, 0, 0, 0.87)"],
      "A700": ["#00bfa5", "rgba(0, 0, 0, 0.87)"]
    },
    GreenT = {
      "50": ["#d0f8ce", "rgba(0, 0, 0, 0.87)"],
      "100": ["#a3e9a4", "rgba(0, 0, 0, 0.87)"],
      "200": ["#72d572", "rgba(0, 0, 0, 0.87)"],
      "300": ["#42bd41", "rgba(0, 0, 0, 0.87)"],
      "400": ["#2baf2b", "rgba(0, 0, 0, 0.87)"],
      "500": ["#259b24", "#ffffff"],
      "600": ["#0a8f08", "#ffffff"],
      "700": ["#0a7e07", "#ffffff"],
      "800": ["#056f00", "rgba(255, 255, 255, 0.87)"],
      "900": ["#0d5302", "rgba(255, 255, 255, 0.87)"],
      "A100": ["#a2f78d", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#5af158", "rgba(0, 0, 0, 0.87)"],
      "A400": ["#14e715", "rgba(0, 0, 0, 0.87)"],
      "A700": ["#12c700", "rgba(0, 0, 0, 0.87)"]
    },
    LightGreenT = {
      "50": ["#f1f8e9", "rgba(0, 0, 0, 0.87)"],
      "100": ["#dcedc8", "rgba(0, 0, 0, 0.87)"],
      "200": ["#c5e1a5", "rgba(0, 0, 0, 0.87)"],
      "300": ["#aed581", "rgba(0, 0, 0, 0.87)"],
      "400": ["#9ccc65", "rgba(0, 0, 0, 0.87)"],
      "500": ["#8bc34a", "rgba(0, 0, 0, 0.87)"],
      "600": ["#7cb342", "rgba(0, 0, 0, 0.87)"],
      "700": ["#689f38", "rgba(0, 0, 0, 0.87)"],
      "800": ["#558b2f", "#ffffff"],
      "900": ["#33691e", "#ffffff"],
      "A100": ["#ccff90", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#b2ff59", "rgba(0, 0, 0, 0.87)"],
      "A400": ["#76ff03", "rgba(0, 0, 0, 0.87)"],
      "A700": ["#64dd17", "rgba(0, 0, 0, 0.87)"]
    },
    LimeT = {
      "50": ["#f9fbe7", "rgba(0, 0, 0, 0.87)"],
      "100": ["#f0f4c3", "rgba(0, 0, 0, 0.87)"],
      "200": ["#e6ee9c", "rgba(0, 0, 0, 0.87)"],
      "300": ["#dce775", "rgba(0, 0, 0, 0.87)"],
      "400": ["#d4e157", "rgba(0, 0, 0, 0.87)"],
      "500": ["#cddc39", "rgba(0, 0, 0, 0.87)"],
      "600": ["#c0ca33", "rgba(0, 0, 0, 0.87)"],
      "700": ["#afb42b", "rgba(0, 0, 0, 0.87)"],
      "800": ["#9e9d24", "rgba(0, 0, 0, 0.87)"],
      "900": ["#827717", "#ffffff"],
      "A100": ["#f4ff81", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#eeff41", "rgba(0, 0, 0, 0.87)"],
      "A400": ["#c6ff00", "rgba(0, 0, 0, 0.87)"],
      "A700": ["#aeea00", "rgba(0, 0, 0, 0.87)"]
    },
    YellowT = {
      "50": ["#fffde7", "rgba(0, 0, 0, 0.87)"],
      "100": ["#fff9c4", "rgba(0, 0, 0, 0.87)"],
      "200": ["#fff59d", "rgba(0, 0, 0, 0.87)"],
      "300": ["#fff176", "rgba(0, 0, 0, 0.87)"],
      "400": ["#ffee58", "rgba(0, 0, 0, 0.87)"],
      "500": ["#ffeb3b", "rgba(0, 0, 0, 0.87)"],
      "600": ["#fdd835", "rgba(0, 0, 0, 0.87)"],
      "700": ["#fbc02d", "rgba(0, 0, 0, 0.87)"],
      "800": ["#f9a825", "rgba(0, 0, 0, 0.87)"],
      "900": ["#f57f17", "rgba(0, 0, 0, 0.87)"],
      "A100": ["#ffff8d", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#ffff00", "rgba(0, 0, 0, 0.87)"],
      "A400": ["#ffea00", "rgba(0, 0, 0, 0.87)"],
      "A700": ["#ffd600", "rgba(0, 0, 0, 0.87)"]
    },
    AmberT = {
      "50": ["#fff8e1", "rgba(0, 0, 0, 0.87)"],
      "100": ["#ffecb3", "rgba(0, 0, 0, 0.87)"],
      "200": ["#ffe082", "rgba(0, 0, 0, 0.87)"],
      "300": ["#ffd54f", "rgba(0, 0, 0, 0.87)"],
      "400": ["#ffca28", "rgba(0, 0, 0, 0.87)"],
      "500": ["#ffc107", "rgba(0, 0, 0, 0.87)"],
      "600": ["#ffb300", "rgba(0, 0, 0, 0.87)"],
      "700": ["#ffa000", "rgba(0, 0, 0, 0.87)"],
      "800": ["#ff8f00", "rgba(0, 0, 0, 0.87)"],
      "900": ["#ff6f00", "rgba(0, 0, 0, 0.87)"],
      "A100": ["#ffe57f", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#ffd740", "rgba(0, 0, 0, 0.87)"],
      "A400": ["#ffc400", "rgba(0, 0, 0, 0.87)"],
      "A700": ["#ffab00", "rgba(0, 0, 0, 0.87)"]
    },
    OrangeT = {
      "50": ["#fff3e0", "rgba(0, 0, 0, 0.87)"],
      "100": ["#ffe0b2", "rgba(0, 0, 0, 0.87)"],
      "200": ["#ffcc80", "rgba(0, 0, 0, 0.87)"],
      "300": ["#ffb74d", "rgba(0, 0, 0, 0.87)"],
      "400": ["#ffa726", "rgba(0, 0, 0, 0.87)"],
      "500": ["#ff9800", "rgba(0, 0, 0, 0.87)"],
      "600": ["#fb8c00", "rgba(0, 0, 0, 0.87)"],
      "700": ["#f57c00", "rgba(0, 0, 0, 0.87)"],
      "800": ["#ef6c00", "#ffffff"],
      "900": ["#e65100", "#ffffff"],
      "A100": ["#ffd180", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#ffab40", "rgba(0, 0, 0, 0.87)"],
      "A400": ["#ff9100", "rgba(0, 0, 0, 0.87)"],
      "A700": ["#ff6d00", "rgba(0, 0, 0, 0.87)"]
    },
    DeepOrangeT = {
      "50": ["#fbe9e7", "rgba(0, 0, 0, 0.87)"],
      "100": ["#ffccbc", "rgba(0, 0, 0, 0.87)"],
      "200": ["#ffab91", "rgba(0, 0, 0, 0.87)"],
      "300": ["#ff8a65", "rgba(0, 0, 0, 0.87)"],
      "400": ["#ff7043", "rgba(0, 0, 0, 0.87)"],
      "500": ["#ff5722", "#ffffff"],
      "600": ["#f4511e", "#ffffff"],
      "700": ["#e64a19", "#ffffff"],
      "800": ["#d84315", "#ffffff"],
      "900": ["#bf360c", "#ffffff"],
      "A100": ["#ff9e80", "rgba(0, 0, 0, 0.87)"],
      "A200": ["#ff6e40", "rgba(0, 0, 0, 0.87)"],
      "A400": ["#ff3d00", "#ffffff"],
      "A700": ["#dd2c00", "#ffffff"]
    },
    BrownT = {
      "50": ["#efebe9", "rgba(0, 0, 0, 0.87)"],
      "100": ["#d7ccc8", "rgba(0, 0, 0, 0.87)"],
      "200": ["#bcaaa4", "rgba(0, 0, 0, 0.87)"],
      "300": ["#a1887f", "#ffffff"],
      "400": ["#8d6e63", "#ffffff"],
      "500": ["#795548", "rgba(255, 255, 255, 0.87)"],
      "600": ["#6d4c41", "rgba(255, 255, 255, 0.87)"],
      "700": ["#5d4037", "rgba(255, 255, 255, 0.87)"],
      "800": ["#4e342e", "rgba(255, 255, 255, 0.87)"],
      "900": ["#3e2723", "rgba(255, 255, 255, 0.87)"]
    },
    GreyT = {
      "50": ["#fafafa", "rgba(0, 0, 0, 0.87)"],
      "100": ["#f5f5f5", "rgba(0, 0, 0, 0.87)"],
      "200": ["#eeeeee", "rgba(0, 0, 0, 0.87)"],
      "300": ["#e0e0e0", "rgba(0, 0, 0, 0.87)"],
      "400": ["#bdbdbd", "rgba(0, 0, 0, 0.87)"],
      "500": ["#9e9e9e", "rgba(0, 0, 0, 0.87)"],
      "600": ["#757575", "rgba(255, 255, 255, 0.87)"],
      "700": ["#616161", "rgba(255, 255, 255, 0.87)"],
      "800": ["#424242", "rgba(255, 255, 255, 0.87)"],
      "900": ["#212121", "rgba(255, 255, 255, 0.87)"],
      "1000": ["#ffffff", "rgba(0, 0, 0, 0.87)"]
    },
    BlueGreyT = {
      "50": ["#eceff1", "rgba(0, 0, 0, 0.87)"],
      "100": ["#cfd8dc", "rgba(0, 0, 0, 0.87)"],
      "200": ["#b0bec5", "rgba(0, 0, 0, 0.87)"],
      "300": ["#90a4ae", "rgba(0, 0, 0, 0.87)"],
      "400": ["#78909c", "#ffffff"],
      "500": ["#607d8b", "#ffffff"],
      "600": ["#546e7a", "rgba(255, 255, 255, 0.87)"],
      "700": ["#455a64", "rgba(255, 255, 255, 0.87)"],
      "800": ["#37474f", "rgba(255, 255, 255, 0.87)"],
      "900": ["#263238", "rgba(255, 255, 255, 0.87)"]
    };

final List<Map<String, String>> colorPalette = [Red, Pink, Purple, DeepPurple, Indigo, Blue, LightBlue, Cyan, Teal, Green, LightGreen, Lime, Yellow, Amber, Orange, DeepOrange, Brown, Grey, BlueGrey];

final List<Map<String, List<String>>> colorPaletteT = [RedT, PinkT, PurpleT, DeepPurpleT, IndigoT, BlueT, LightBlueT, CyanT, TealT, GreenT, LightGreenT, LimeT, YellowT, AmberT, OrangeT, DeepOrangeT, BrownT, GreyT, BlueGreyT];

Map getRandomColorPalette() {
  Random rng = new Random();
  num randomColorId = rng.nextInt(colorPalette.length);
  Map randomColor = colorPalette.elementAt(randomColorId);
  return randomColor;
}

Map getRandomColorPaletteT() {
  Random rng = new Random();
  num randomColorId = rng.nextInt(colorPaletteT.length);
  Map randomColorT = colorPaletteT.elementAt(randomColorId);
  return randomColorT;
}

String getRandomColor(String level) {
  return getRandomColorPalette()[level];
}

List getRandomColorT(String level) {
  return getRandomColorPaletteT()[level];
}
//for manual extraction of color via console window on http://www.google.com/design/spec/style/color.html
/*
javascriptFunctionToCrawLColorsFromHttpWwwGoogleComDesignSpecStyleColorHtmlConsole() {
  palette = document.getElementsByClassName("color-palette")[0].children;
  colorGroupsList = [];
  var JSONcolorGroup = {};
  for (var i = 0; i < palette.length; i++) {
    var colorGroup = palette[i].children[0];
    var colors = colorGroup.children;
    var name = colors[0].children[0].innerHTML.replace(" ", "");
    var textColor = "";
    //var JSONcolors = [] ;
    var JSONcolor = {};
    for (var j = 1; j < colors.length; j++) {
      if(colors[j] && colors[j].className.indexOf("light") > -1){
        textColor = "#ffffff"; 
      }else if(colors[j] && colors[j].className.indexOf("dark") > -1){
        textColor = "rgba(0, 0, 0, 0.87)";
      }else{
        textColor = "rgba(255, 255, 255, 0.87)";
      }
      
      var key = colors[j].children[0].innerHTML.toString().replace(" ", "");
      var color = colors[j].children[1].innerHTML;
      var all = [color,textColor]; 
      eval("JSONcolor['" + key + "'] = all");
    }
    eval("JSONcolorGroup['" + name + "']=JSONcolor;");
    JSON.stringify(JSONcolorGroup);
  }
  }
*/
